cpp/src/parquet/arrow/writer.cc

```diff
@@ -176,18 +175,17 @@ class LevelBuilder {
       }
       *num_levels = array.length();
     } else {
+      // Note it is hard to estimate memory consumption due to zero length
+      // arrays otherwise we would preallocate.  An upper boun on memory
+      // is the sum of the length of each list array  + number of elements
+      // but this might be too loose of an upper bound so we choose to use
+      // safe methods.
       RETURN_NOT_OK(rep_levels_.Append(0));
       RETURN_NOT_OK(HandleListEntries(0, 0, 0, array.length()));
 
-      std::shared_ptr<Array> def_levels_array;
-      std::shared_ptr<Array> rep_levels_array;
-
-      RETURN_NOT_OK(def_levels_.Finish(&def_levels_array));
-      RETURN_NOT_OK(rep_levels_.Finish(&rep_levels_array));
-
-      *def_levels_out = static_cast<PrimitiveArray*>(def_levels_array.get())->values();
-      *rep_levels_out = static_cast<PrimitiveArray*>(rep_levels_array.get())->values();
-      *num_levels = rep_levels_array->length();
+      RETURN_NOT_OK(def_levels_.Finish(def_levels_out));
```

Comment:
Nice simplification here

Reply:
The CI failures look spurious: Travis is in a re-run cmake loop and appveyor failed to build googletest